src/test/java/kr/codesqaud/cafe/largest/UserErrorLargeTest.java

```diff
@@ -0,0 +1,80 @@
+package kr.codesqaud.cafe.largest;
+
+import static org.assertj.core.api.AssertionsForClassTypes.*;
+import static org.junit.jupiter.api.DynamicTest.*;
+
+import java.util.stream.Stream;
+
+import javax.servlet.http.HttpSession;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.DynamicTest;
+import org.junit.jupiter.api.TestFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+
+import kr.codesqaud.cafe.domain.auth.controller.AuthController;
+import kr.codesqaud.cafe.domain.auth.dto.request.AuthLoginRequestDto;
+import kr.codesqaud.cafe.domain.user.controller.UserController;
+import kr.codesqaud.cafe.domain.user.dto.request.UserSaveRequestDto;
+import kr.codesqaud.cafe.domain.user.dto.request.UserUpdateRequestDto;
+import kr.codesqaud.cafe.domain.user.entity.User;
+import kr.codesqaud.cafe.domain.user.repository.UserRepository;
+import kr.codesqaud.cafe.global.exception.user.DuplicatedUserIdException;
+import kr.codesqaud.cafe.global.exception.user.InvalidLoginInfoException;
+import kr.codesqaud.cafe.global.exception.user.InvalidPasswordException;
+
+@SpringBootTest
+public class UserErrorLargeTest {
+
+	@Autowired
+	private UserRepository userRepository;
+
+	@Autowired
+	private UserController userController;
+
+	@Autowired
+	private AuthController authController;
+
+	@Autowired
+	HttpSession httpSession;
+
+	@DisplayName("회원가입 ,로그인, 수정 시 오류")
+	@TestFactory
+	Stream<DynamicTest> userError() {
+		return Stream.of(
+			dynamicTest("회원 가입 시 동일한 ID가 있을 경우", () -> {
+				//given
+				userRepository.save(new User("pjs970616", "June", "1234", "pjs970616@naver.com"));
+
+				//when & then
+				assertThatThrownBy(() -> userController.signIn(
+					new UserSaveRequestDto("pjs970616", "Joonsoo", "1234", "Joonsoo@naver.com"))).isInstanceOf(
+					DuplicatedUserIdException.class);
+
+			}),
+			dynamicTest("로그인시 올바르지 않은 아이디를 입력한 경우", () -> {
+				//when & then
+				assertThatThrownBy(() ->
+					authController.login(new AuthLoginRequestDto("wrongId", "1234"),
+						httpSession)).isInstanceOf(
+					InvalidLoginInfoException.class);
+			}),
+			dynamicTest("로그인시 올바르지 않은 비밀번호를 입력한 경우", () -> {
+				//when & then
+				assertThatThrownBy(() ->
+					authController.login(new AuthLoginRequestDto("pjs970616", "wrongPassword"),
+						httpSession)).isInstanceOf(
+					InvalidLoginInfoException.class);
+			}),
```

Comment:
```suggestion
                dynamicTest("로그인시 올바르지 않은 비밀번호를 입력한 경우", () -> {
                    //when & then
                    AuthLoginRequestDto requestDto = new AuthLoginRequestDto("pjs970616", "wrongPassword");
                    assertThatThrownBy(() -> authController.login(requestDto, httpSession))
                            .isInstanceOf(InvalidLoginInfoException.class);
                }),
```

좋은 가독성에 대해서는 개인차가 있을 수 있습니다.
따라서 리팩토링에서 제가 고려한 점을 전달 드립니다.

- 매개 변수가 너무 많은 것이 아니라면, 매개 변수를 개행하지 않습니다.
- dto 는 사전에 생성하여 로직에만 집중합니다.

Reply:
코드 짤 때 참고하겠습니다. 수정하겠습니다